pkg/alertmanager/state_replication.go

```diff
@@ -262,6 +262,23 @@ func (s *state) Ready() bool {
 	return s.Service.State() == services.Running
 }
 
+func (s *state) MergeGrafanaState(fs []*clusterpb.FullState) error {
+	if err := s.MergeFullStates(fs); err != nil {
+		return err
+	}
+
+	for _, fs := range fs {
```

Comment:
Feels like we should have a lock around this whole process but `MergeFullStates` already locks internally.

Why not add the broadcasting directly to `MergeFullStates`? This is technically a bug in Cloud AM to, even though there aren't really many (any?) current avenues where new states are created directly in storage without calling  `setSilence`.

Reply:
I don't think there are any ways we could be getting a silence from the state without it being created via `setSilence()` before. This is only possible for Grafana AM silences, so I'd avoid changing how merging full state works for all tenants.